gift-mode.el

```diff
@@ -42,6 +42,50 @@
 (require 'outline)
 (require 'newcomment)
 
+
+(defvar gift-imenu-comments-regexp "\\(\\(^\\s-*//.*$\\)\\|\\(^\\s-*$\\)\\)+")
+
+(defvar gift-imenu-question-regexp "^\\(\\([^{}]\\)+\\){" )
+
+(defvar gift-imenu-title-regexp "::\\(.*\\)::" )
+
+
+(defun gift-imenu-log (msg &rest args)
+  "Utility log function, same as message."
+  (when nil
+    (apply #'message msg args)))
+
+(defun gift-imenu-sanitize-title (title)
+  "Cleans a question: empty lines, comments, spaces, extract title."
+  (let*
+      ((nocoments (replace-regexp-in-string gift-imenu-comments-regexp " " (match-string 1) nil 'literal) )
+       (noextrablanks (replace-regexp-in-string "\\(\n\\|\\s-\\)+" " " nocoments nil 'literal) )
+       (trimmed (trim-string noextrablanks)))
+    (if (string-match gift-imenu-title-regexp trimmed)
+        (match-string 1 trimmed)
+      trimmed)))
+  
+
+(defun gift-imenu-index()
+  "Return a table of contents for a gift buffer for use with Imenu."
+  (goto-char (point-min))
+  (setq retorno (list) )
```

Comment:
I know in the next commit you've bound these variables.  My Spanish is worse than your English, I'm sure, and it's unfair of me to ask, but I think this would be easier in general to cope with if they were renamed retorno => result, posicion => position, entrada => entry and nombre => index (or similar).

Reply:
You are right.
Learning elisp and translating at the same time is so difficult :)